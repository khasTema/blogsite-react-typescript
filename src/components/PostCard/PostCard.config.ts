export interface IPostCardConfig {
    moreLink: string;
}

export const postCardConfig: IPostCardConfig = {
    moreLink: 'read more'
}