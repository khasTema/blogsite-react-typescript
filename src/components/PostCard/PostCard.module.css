.card {
    width: 90%;
    border: var(--standart-border);
    border-radius: 10px;
    padding: 1rem;
    margin-bottom: 1rem;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}

.card h3 {
   font-family: var(--headings-font);
}

.card p {
    font-family: var(--text-font);
    font-weight: var(--fontweight-reg);
    color: var(--text-secondary-color);
}

.info {
    display: flex;
    justify-content: space-between;
}

.more {
    display: flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    text-transform: uppercase;
    font-family: var(--text-font);
    font-size: 1rem;
    font-weight: var(--fontweight-semibold);
    color: var(--logo-text-color);
    background-color: var(--button-color);
    padding: .3rem;
    opacity: .7;
    transition: all .3s;
}

.more:hover {
    opacity: 1;
}

@media screen and (min-width: 717px) {
    .card {
        width: 47%;
    }
}