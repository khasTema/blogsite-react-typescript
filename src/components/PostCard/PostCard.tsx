import React from 'react'
import { IPost } from '../../interface'
import styles from './PostCard.module.css'
import { postCardConfig } from './PostCard.config'
import { Button } from '../UI/Button'

export const PostCard: React.FC<IPost> = ({userId, title, body, id}):JSX.Element => {
  return (
    <div className={styles.card}>
        <h3>{title}</h3>
        <p>{body}</p>
        <div className={styles.info}>
            <span className={styles.user}>{userId}</span>
            <Button linkTo={`/details/${id}`} textBtn={postCardConfig.moreLink}/>
        </div>
    </div>
  )
}
