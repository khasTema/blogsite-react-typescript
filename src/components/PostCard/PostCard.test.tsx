import renderer from 'react-test-renderer';
import React from 'react'
import { PostCard } from './PostCard';
import { JSX } from 'react/jsx-runtime';

jest.mock('../../interface');
jest.mock('./PostCard.module.css');
jest.mock('./PostCard.config');
jest.mock('../UI/Button');

const renderTree = (tree: JSX.Element) => renderer.create(tree);
describe('<PostCard>', () => {
  it('should render component', () => {
    expect(renderTree(<PostCard userId={0} id={0} title={''} body={''}/>).toJSON()).toMatchSnapshot();
  });
  
});