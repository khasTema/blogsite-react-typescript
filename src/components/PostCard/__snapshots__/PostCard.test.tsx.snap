// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard> should render component 1`] = `
<div>
  <h3 />
  <p />
  <div>
    <span>
      0
    </span>
  </div>
</div>
`;
